Migration guide for SnackBar default behavior with action (#12329)
flutter/website: src/content/release/breaking-changes/snackbar-with-action-behavior-update.md

---
title: SnackBar with action no longer auto-dismisses
description: >-
  SnackBars with an action button now default to not auto-dismissing unless
  manually dismissed by the user.
---

## Summary

The default behavior of a [`SnackBar`][] with an action has changed. Previously, a

> `SnackBar` with an action would only not auto-dismiss if talkback was enabled. 

Reviewer: ```suggestion
`SnackBar` with an action would not auto-dismiss if talkback was enabled. 
```

The "only" added confusion, for me, at least. Too many negatives, in effect. But, does this make the text wrong?

Author: Removing it still sounds correct to me. Thank you for the improvement!

Now, all `SnackBar`s with an action default to a non-dismissible state until the
user interacts with the action button.

## Context

A `SnackBar` with an action button is now treated as a more persistent
notification that requires user interaction. This change improves accessibility
and user experience by ensuring that critical notifications remain on the screen
until they are acknowledged.

## Description of change

This change aligns with the Material 3 design specifications for `SnackBar`s:

> * Old behavior: A `SnackBar` with an action button would auto-dismiss after a
> duration unless talkback was enabled.
> * New behavior: A `SnackBar` with an action button will not auto-dismiss. It will 
> remain on screen until it is dismissed by the user.

Reviewer: ```suggestion
* Old behavior: A `SnackBar` with an action button would auto-dismiss after a
  duration unless talkback was enabled.
* New behavior: A `SnackBar` with an action button won't auto-dismiss; 
   it remains on screen until dismissed by the user.
```

To override this behavior, an optional `persist` property has 
been added to `SnackBar`. When `persist` is true, the `SnackBar` won't auto-dismiss  
and remains on screen until manually dismissed by the user. When false, the 
`SnackBar` auto-dismisses after its standard duration, regardless of the 
presence of an action. When null, the `SnackBar` follows the default 

> behavior, which won't auto-dismiss if an action is present.

Reviewer: It rather saddens me that we add something to help with accessibility and then provide a way to undo it...

> To allow users to override this behavior, an optional `persist` property has 
> been added to `SnackBar`. When it is true, the `SnackBar` will not auto-dismiss  
> and will remain on screen until manually dismissed by the user. When false, the 
> `SnackBar` will auto-dismiss after its standard duration, regardless of the 
> presence of an action. When it is null, the `SnackBar` will use the default 
> behavior, which is to not auto-dismiss if an action is present.

Reviewer: ```suggestion
To override this behavior, an optional `persist` property has 
been added to `SnackBar`. When `persist` is true, the `SnackBar` won't auto-dismiss  
and remains on screen until manually dismissed by the user. When false, the 
`SnackBar` auto-dismisses after its standard duration, regardless of the 
presence of an action. When null, the `SnackBar` follows the default 
behavior, which won't auto-dismiss if an action is present.
```
